src/modules/Rotate/Rotate.js

```diff
@@ -0,0 +1,46 @@
+module.exports = function Rotate(pixels, pixels2, options, rotate_value, width, height, cos, sin){
+  var imagejs = require('imagejs');
+  var height_half = Math.floor(height / 2);
+  var width_half = Math.floor(width / 2);
+  var dimension = width + height;
+
+  if (rotate_value % 360 == 0)
+    return pixels;
+  function copyPixel(x1, y1, x2, y2){
+    pixels1.set(x1, y1, 0, pixels.get(x2, y2, 0));
+    pixels1.set(x1, y1, 1, pixels.get(x2, y2, 1));
+    pixels1.set(x1, y1, 2, pixels.get(x2, y2, 2));
+    pixels1.set(x1, y1, 3, pixels.get(x2, y2, 3));
+  }
+
+  function copyPixel2(x1, y1, x2, y2){
+    pixels2.set(x1, y1, 0, pixels1.get(x2, y2, 0));
+    pixels2.set(x1, y1, 1, pixels1.get(x2, y2, 1));
+    pixels2.set(x1, y1, 2, pixels1.get(x2, y2, 2));
+    pixels2.set(x1, y1, 3, pixels1.get(x2, y2, 3));
+  }
+
+  pixels1 = require('ndarray')(new Uint8Array(4 * dimension * dimension).fill(0), [dimension, dimension, 4]);
+  //copying all the pixels from image to pixels1
+  for (var n = 0; n < pixels.shape[0]; n++){
+    for (var m = 0; m < pixels.shape[1]; m++){
+      copyPixel(n + height_half, m + width_half, n, m);
+    }
+  }
+  //rotating pixels1
+  var bitmap = new imagejs.Bitmap({ width: pixels1.shape[0], height: pixels1.shape[1] });
+  bitmap._data.data = pixels1.data;
+
+  var rotated = bitmap.rotate({
```

Comment:
Hmm, so why do we need to do so much pixel copying and rotation math if we are still using the Imagejs Bitmap function to perform the rotation? Perhaps some comments would help clarify what's going on here? Sorry, it's just a little hard to follow! Thanks for your help, i love the output, just being sure I understand what's happening inside as well :-)

Reply:
earlier if we use bitmap on original image then it cropped some of the image.to solve this i have put the image on a larger frame ie is square  so that there is no problem of clipping and then extra white is cropped according to to final image dimensions after rotation

Reply:
pixel1 is a square frame with original image at it centre,then pixel1 is rotated with the help of imagejs.at last we have have copied pixels from pixels1 to pixel2 so that there is no whitespace.

Reply:
pixel1 is a square frame that contain original image at it's centre.the we rotated the image with help of imagejs,at the last we copied pixels from pixel1 to pixel2 so that there is no extra whitespace.